tests/framework/utils/certificate_utils.go

```diff
@@ -0,0 +1,78 @@
+package utils
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"math/big"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+)
+
+// Webhook generation approach from https://gist.github.com/velotiotech/2e0cfd15043513d253cad7c9126d2026#file-initcontainer_main-go
+
+// GenerateWebhookCerts will return the CA certificate, the Server certificate, and the Server private key
+func GenerateWebhookCerts(s suite.Suite, dnsNames []string) (*bytes.Buffer, *bytes.Buffer, *bytes.Buffer) {
+	ca := &x509.Certificate{
+		SerialNumber: big.NewInt(2222),
+		Subject: pkix.Name{
+			Organization: []string{"Mondoo Operator E2E CA"},
+		},
+		NotBefore: time.Now(),
+		NotAfter:  time.Now().AddDate(0, 0, 5),
+		IsCA:      true,
+		//ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		BasicConstraintsValid: true,
+	}
+
+	caPrivKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	s.NoError(err, "Failed to generate private key for CA")
```

Comment:
```suggestion
	s.NoErrorf(err, "Failed to generate private key for CA")
```

You should use `NoErrorf` everywhere you want to provide a specific error message. I saw that you used `NoError` in multiple places. I won't add a separate comment for each. Perhaps just go through the assertions once more and change where needed